tests/transformations/test_transformations.py

```diff
@@ -96,3 +167,119 @@ def extract_test(dataset: SupportsReadableDataset[Any]) -> Any:
+    fruit_p.run(fruitshop_source())
+
+    @dlt.transformation()
+    def materializable_sql_model(dataset: Dataset) -> Any:
+        yield "SELECT id, name FROM customers"
+
+    model = list(materializable_sql_model(fruit_p.dataset()))[0]
+    assert model.arrow().column_names == ["id", "name"]
+    assert model.schema["columns"] == {
+        "id": {"name": "id", "data_type": "bigint", "nullable": False},
+        "name": {"name": "name", "x-annotation-pii": True, "data_type": "text", "nullable": True},
+    }
+
+
+@pytest.mark.skipif(sys.version_info < (3, 10), reason="Ibis Requires Python 3.10 or higher")
+@pytest.mark.parametrize(
+    "destination_config",
+    transformation_configs(only_duckdb=True),
+    ids=lambda x: x.name,
+)
+def test_ibis_unbound_table_transformation(
+    destination_config: DestinationTestConfiguration,
+) -> None:
+    fruit_p, dest_p = setup_transformation_pipelines(destination_config)
+    fruit_p.run(fruitshop_source())
+
+    @dlt.transformation()
+    def materializable_sql_model(dataset: Dataset) -> Any:
+        purchases = dataset.table("purchases", table_type="ibis")
+        customers = dataset.table("customers", table_type="ibis")
+        yield purchases.join(customers, purchases.customer_id == customers.id)[
+            ["id", "customer_id", "inventory_id", "quantity", "name"]
+        ]
+
+    model = list(materializable_sql_model(fruit_p.dataset()))[0]
+    assert model.arrow().column_names == [
+        "id",
+        "customer_id",
+        "inventory_id",
+        "quantity",
+        "name",
+    ]
+    assert model.arrow().shape == (3, 5)
+
+
+@pytest.mark.parametrize(
+    "destination_config",
+    transformation_configs(only_duckdb=True),
+    ids=lambda x: x.name,
+)
+def test_transformations_without_generator(
+    destination_config: DestinationTestConfiguration,
+) -> None:
+    fruit_p, dest_p = setup_transformation_pipelines(destination_config)
+    fruit_p.run(fruitshop_source())
+
+    @dlt.transformation()
+    def transform_without_generator(dataset: Dataset) -> Any:
+        # return works!
+        return dataset["customers"]
+
+    dest_p.run(transform_without_generator(fruit_p.dataset()))
+
+    assert load_table_counts(dest_p, "transform_without_generator") == {
+        "transform_without_generator": 13,
+    }
+
+    assert get_job_types(dest_p) == {"transform_without_generator": {"model": 1}}
+
+
+# @pytest.mark.skip("TODO: hints are not being merged correctly")
+
+
+# TODO: support multiple transformations in a single function
+@pytest.mark.parametrize(
+    "destination_config",
+    transformation_configs(only_duckdb=True),
+    ids=lambda x: x.name,
+)
+def test_multiple_transformations_in_function(
+    destination_config: DestinationTestConfiguration,
+) -> None:
+    fruit_p, dest_p = setup_transformation_pipelines(destination_config)
+    fruit_p.run(fruitshop_source())
+
+    @dlt.transformation()
+    def multiple_transformations(dataset: Dataset) -> Any:
```

Comment:
would be cool to enumerate resource without pipeline and also to have a mixed easy and eager transformations yielded